sweepbatcher/sweep_batch.go

```diff
@@ -1314,15 +1334,78 @@ func constructUnsignedTx(sweeps []sweep, address btcutil.Address,
 		feeForWeight++
 	}
 
+	// Add the batch transaction output, which excludes the fees paid to
+	// miners. Reduce the amount by the sum of change outputs, if any.
+	var sumChange int64
+	for _, change := range changeOutputs {
+		sumChange += change.Value
+	}
+
+	// Ensure that the batch amount is greater than the sum of change.
+	if batchAmt <= btcutil.Amount(sumChange) {
+		return nil, 0, 0, 0, fmt.Errorf("batch amount %v is <= the "+
+			"sum of change outputs %v", batchAmt,
+			btcutil.Amount(sumChange))
+	}
+
 	// Clamp the calculated fee to the max allowed fee amount for the batch.
-	fee := clampBatchFee(feeForWeight, batchAmt)
+	fee := clampBatchFee(feeForWeight, batchAmt-btcutil.Amount(sumChange))
```

Comment:
In test `TestPresigned/dust_main_output` I saw:

> fee 0.00000065 BTC

Such a fee is too low (below relay fee).

I think we should fail here if the fee is below relay fee rate. The function should accept min relay fee rate as an argument, we already have the value in many places, where the function is called.

Feerate can be found with `chainfee.NewSatPerKWeight(fee, weight)`.

Please add a unit test for `constructUnsignedTx` for this error and adjust test `TestPresigned/dust_main_output` not to crash on this error. Additionally, you can add another test case for TestPresigned to reproduce this new error (fee below relay fee).

Reply:
I've threaded the min relay fee through to `clampBatchFee` which now checks it the tx fee is below the min relay fee, and errors if so.

I've also added a test case for that scenario and fixed `TestPresigned/dust_main_output` by lowering lnd's min relay fee in that test.